.github/workflows/swift_package_test.yml

```diff
@@ -1,8 +1,24 @@
-name: Swift Linux Matrix
+name: Swift Matrix
 
 on:
   workflow_call:
     inputs:
+      macos_xcode_versions:
+        type: string
+        description: "Exclude Xcode version list (JSON)"
+        default: "[\"16.0\", \"16.1\", \"16.2\"]"
+      macos_exclude_xcode_versions:
+        type: string
+        description: "Exclude Xcode version list (JSON)"
+        default: "[{\"xcode_version\": \"\"}]"
+      macos_versions:
+        type: string
+        description: "macOS version list (JSON)"
+        default: "[\"sequoia\"]"
```

Comment:
I personally would have probably chosen the release version number (eg. `15` for Sequioa) instead of the marketing name but don’t have strong opinions here. Also just checking: We don’t want to add minor-version fidelity here (eg. the ability to select 15.4 instead of 15.0 or something like that)?

Finally, I think this assumes that we also have a self-hosted runner for this arch + macOS version combination, right? Maybe worth pointing that out in a comment, eg. something like `This requires a self-hosted macOS runner with this version to be set up.` 

If we don’t have near-term plans to support multiple OSs and arches, I would probably remove the option to not give users a false sense that they can test all macOS + arch + Xcode versions.